ProcessManager.py

```diff
@@ -3,50 +3,97 @@
+from multiprocessing.context import Process # Processes are fully functional on Windows.
 import time
 import sys
+from subprocess import Popen
+
+class RoboProcesses:
+    def __init__(self, cmd):
+        self.cmd = cmd
+        self.popen = None
+    def execute(self):
+        self.popen = Popen(self.cmd)
+
+
 
 class ProcessManager:
     """
     Class: ProcessManager
-    
+
     This class represents the pocessing module for the robocluster operating system.
     """
-    
+    process_dict = {}
+
+
     def __init__(self, **kwargs):
         """Initializes the process manager"""
-        self.T = [] # Thread or process library (It is right now a list but it could be changed to something else other than a list).
+        self.process_dict = {} # Thread or process library (It is right now a list but it could be changed to something else other than a list).
         return super().__init__(**kwargs)
 
     def isEmpty(self):
-        if T == []:
+        """
+        Confirms whether process_dict is empty.
+        """
+        if self.process_dict == {}:
             return True
         else:
             return False
 
-    def createProcess(self):
+    def createProcess(self, name, command):
         """
-        TODO create a function that takes in parameters and uses those parameters to
-        ceate one single process. The process must then be stored in "self.T".
+        Creates a process.
+        Checks if command is type string if so it returns True otherwise it returns False.
+        Checks if name has not been used otherwise informs that name is taken.
+
         """
+        if type(command) == str:
+            for process in self.process_dict:
+                if process == name:
+                    print ("Name already taken.")
+                    break
+            else:
+                self.process_dict[name] = RoboProcesses(command)
+            return True
+        else:
+            return False
+
+
+
+
 
     def createThread(self):
         """
         TODO create a function that takes in parameters and uses those parameters to
         ceate one single thread. The thread must then be stored in "self.T".
         """
 
+    def startProcess(self, name):
+        """
+        Starts a single process.
+        """
+        self.process_dict[name].execute()
+
+
     def startAllProcesses(self):
         """
-        TODO: create a function that starts all processes or threads in "self.T" all at once.
+        Starts all processes in process_dict.
         """
 
+        for process in self.process_dict:
+            self.startProcess(process)
+
     def stopAllProcesses(self):
         """
-        TODO: create a function that stops all processes or threads in "self.T" all at once.
+        Stops all processes in process_dict.
         """
 
+        print ("Shutting down")
+
+        for process in self.process_dict:
+            process.execute.kill()
+            process.execute.wait()
```

Comment:
We'll have to access the popen object like so:
```python
for process in self.process_dict:
    self.process_dict[process].popen.kill()
    ...
```

Reply:
Also could we add a method similar to `startProcess` that stops a single process? It would be usefull later on when restarting processes.